Add Fusion as primary use case example (#3741)
nextflow-io/nextflow: docs/wave.rst

The first repository is used to store the built container images. The second one is used to store the individual image layers for caching purposes.

> The repository access keys need to be provided as Tower credentials (see

Author: ```suggestion
The repository access keys must be provided as Tower credentials (see
```
